src/hotspot/share/oops/markWord.hpp

```diff
@@ -240,8 +240,11 @@ class markWord {
   }
   bool has_displaced_mark_helper() const {
     intptr_t lockbits = value() & lock_mask_in_place;
-    return LockingMode == LM_LIGHTWEIGHT  ? UseObjectMonitorTable ? false : lockbits == monitor_value   // no displace mark or monitor?
-                                         : (lockbits & unlocked_value) == 0; // monitor | stack-locked?
+    if (LockingMode == LM_LIGHTWEIGHT) {
```

Comment:
Take this if you want. There were some extra whitespaces, and the nested ternaries are confusing at best.